core/src/banking_stage/consumer.rs

```diff
@@ -460,7 +460,7 @@ impl Consumer {
         pre_results: impl Iterator<Item = Result<(), TransactionError>>,
     ) -> ProcessTransactionBatchOutput {
         let (
-            (transaction_qos_cost_results, cost_model_throttled_transactions_count),
+            (mut transaction_qos_cost_results, cost_model_throttled_transactions_count),
```

Comment:
as an alternative to mutability here. we could also use the lock-results to inform the cost-adjustment post-execution. Not sure if that would be cleaner though.